src/ssa/ssa_dereference.cpp

```diff
@@ -248,6 +248,9 @@ exprt ssa_alias_value(
   }
 
   byte_extract_exprt byte_extract(byte_extract_id(), e1.type());
+#ifdef COMPETITION
```

Comment:
if(options.get_bool_option("competition-mode"))

Reply:
`options` is not available in `local_SSAt`. Should I pass it there? It will require quite a lot of changes since it would have to be passed as a parameter to all constructor calls.

Reply:
You could thread it through ssa_heap_analysist, which might be easier, but clearly uglier, but still better than an obscure combination of compile-time and runtime flags.

Reply:
Ok, in the end, I passed `options` to `local_SSAt` and implemented it correctly. The new commit also gets rid of the `COMPETITION` flag in `ssa_value_set` and so all competition settings can be configured via CLI and no special compilation is needed.